src/moonlink/src/table_handler/chaos_replay.rs

```diff
@@ -90,12 +99,22 @@ pub(crate) async fn replay() {
     // Used to notify certain background table events have completed.
     let event_notification = Arc::new(Notify::new());
     let event_notification_clone = event_notification.clone();
+
     // Current table states.
     let mut ongoing_flush_event_id = HashSet::new();
     let completed_flush_events: Arc<Mutex<HashMap<BackgroundEventId, CompletedFlush>>> =
         Arc::new(Mutex::new(HashMap::new()));
     let completed_flush_events_clone = completed_flush_events.clone();
 
+    let mut ongoing_mooncake_snapshot_id = HashSet::new();
+    let completed_mooncake_snapshots: Arc<
+        Mutex<HashMap<BackgroundEventId, CompletedMooncakeSnapshot>>,
+    > = Arc::new(Mutex::new(HashMap::new()));
+    let completed_mooncake_snapshots_clone = completed_mooncake_snapshots.clone();
+
+    // Expected table snapshot.
+    let mut snapshot_disk_files = HashSet::new();
```

Comment:
I did simple assertion first in the PR, only for disk files, will followup with another PR to assert deletion vectors (which is much harder).